### PR TITLE
LPS-149497 Generate methods to get non included fields in the API model but included in the URI path

### PR DESCRIPTION
**Pull request by ccorreagg**
Related ticket: [LPS-149497](https://issues.liferay.com/browse/LPS-149497)

This PR contains an updated template for the `Base*ResourceTestCase` that includes the auto-generation of new protected methods to allow, from any extended Test, to modify the value of the attributes that are not included in the API model.

Before the PR:

The test should be completely overriden without calling the test from the parent because it was not possible to customize the value of any parameter included in the path of the URL but not in the requested API model.

```
@Test
public void testGetTermOrderTypeOrderType() throws Exception {
        OrderType postOrderType = testGetTermOrderTypeOrderType_addOrderType();
        
        OrderType getOrderType = orderTypeResource.getTermOrderTypeOrderType(null);
        
        assertEquals(postOrderType, getOrderType);
        assertValid(getOrderType);
}
```

After the PR:

It is possible to use the test from the Base class by implementing/overriding the required methods.

```
@Test
public void testGetTermOrderTypeOrderType() throws Exception {
	OrderType postOrderType = testGetTermOrderTypeOrderType_addOrderType();

	OrderType getOrderType = orderTypeResource.getTermOrderTypeOrderType(
		testGetTermOrderTypeOrderType_getTermOrderTypeId());

	assertEquals(postOrderType, getOrderType);
	assertValid(getOrderType);
}
```

```
protected Long testGetTermOrderTypeOrderType_getTermOrderTypeId()
	throws Exception {

	throw new UnsupportedOperationException(
		"This method needs to be implemented");
}
```

### CODE REVIEW

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by ccorreagg** [comment]
ci:reevaluate:1347401_1258

**Comment by ccorreagg** [comment]
ci:test:relevant

File: modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_test_case.ftl [hunk 1]
```diff
@@ -269,7 +271,8 @@ public abstract class Base${schemaName}ResourceTestCase {
 							<#elseif properties?keys?seq_contains(javaMethodParameter.parameterName)>
 								${schemaVarName}.get${javaMethodParameter.parameterName?cap_first}()
 							<#else>
-								null
+								<#assign missingGetterJavaMethodParametersMap = missingGetterJavaMethodParametersMap + {javaMethodParameter.parameterName: javaMethodParameter} />
```
Comment (javierdearcos) [inline modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_test_case.ftl:274]: Wow 😱 

Didn't know we can create a map like this 👏

File: modules/apps/headless/headless-admin-content/headless-admin-content-test/src/testIntegration/java/com/liferay/headless/admin/content/resource/v1_0/test/StructuredContentResourceTest.java [hunk 2]
```diff
@@ -217,6 +183,22 @@ protected StructuredContent randomStructuredContent() throws Exception {
 		return structuredContent;
 	}
 
+	@Override
+	protected StructuredContent
```
Comment (javierdearcos) [inline modules/apps/headless/headless-admin-content/headless-admin-content-test/src/testIntegration/java/com/liferay/headless/admin/content/resource/v1_0/test/StructuredContentResourceTest.java:187]: Great refactor 👏 👏 👏

**Review by javierdearcos — APPROVED** [review]
Good job!

Thanks to you I've learned how to create maps in the templates 😄

**Comment by javierdearcos** [comment]
ci:forward

**Comment by ccorreagg** [comment]
ci:stop

**Comment by ccorreagg** [comment]
There was a modification to avoid updating local variables within macros as the value of the variable is not being modified

**Comment by ccorreagg** [comment]
ci:test:relevant

**Comment by javierdearcos** [comment]
Checked again. Good work!

**Comment by javierdearcos** [comment]
ci:test:sf

**Comment by javierdearcos** [comment]
ci:test:relevant

**Comment by john-co** [comment]
cc/ @zhangyan0701 , please follow up with test analysis if still failing and manually forward.

**Comment by ccorreagg** [comment]
ci:forward

**Comment by ccorreagg** [comment]
> cc/ @zhangyan0701 , please follow up with test analysis if still failing and manually forward.

Thank you John! Finally it worked 😄

**Pull request closed without merging**